frontend/src/app/NavSidebar.tsx

```diff
@@ -14,15 +14,17 @@ const createNavItem = ({ id, label, href }, pathname) => {
 };
 
 const createNavGroup = ({ group, children }, pathname) => {
-  const isActive = !!children.find((c) => pathname.startsWith(c.href));
+  const isActive = !!children.find((c) => pathname === c.href);
+  const [expanded, setExpanded] = React.useState<boolean>(isActive);
```

Comment:
just noting this one, technically it works, but it makes the linter cry about wanting to be a functional component